Drop Custom toJS() / toJSNewlyCreated() functions for SVGPathSeg

cdumez opened the pull request:
#### a50f45d1c65d53d4ab9d50b2bd5432ab523a75b7
<pre>
Drop Custom toJS() / toJSNewlyCreated() functions for SVGPathSeg
<a href="https://bugs.webkit.org/show_bug.cgi?id=300288">https://bugs.webkit.org/show_bug.cgi?id=300288</a>

Reviewed by Said Abou-Hallawa and Sam Weinig.

Drop Custom toJS() / toJSNewlyCreated() functions for SVGPathSeg now that
the bindings generator knows how to generate the right code.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/JSSVGPathSegCustom.cpp: Removed.
* Source/WebCore/svg/SVGPathSeg.h:
* Source/WebCore/svg/SVGPathSeg.idl:
* Source/WebCore/svg/SVGPathSegImpl.h:

Canonical link: <a href="https://commits.webkit.org/301171@main">https://commits.webkit.org/301171@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d648baf1a5e2def99804f6e58379cdb3034ed73e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124984 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44653 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35390 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131833 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76863 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7be11f76-4ea9-4faa-89e2-e439122ce67f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126861 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45345 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53219 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95128 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63117 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; Exiting early after 60 failures. 6426 tests run. 60 failures; Uploaded test results; Exiting early after 60 failures. 6294 tests run. 60 failures; Compiled WebKit; layout-tests (exception)") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5534bcfb-6d14-4700-bd11-37096c75ee60) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127938 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36192 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111780 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75675 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/837b46f7-5a72-4f95-bfbc-7d2d240737ba) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35119 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29933 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75313 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105955 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30164 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134512 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51807 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39608 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103606 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52232 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107993 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103382 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26355 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48722 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27011 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48833 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51691 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57486 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51065 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54422 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52757 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/svg/SVGPathSegImpl.h:
```diff
@@ -222,3 +222,23 @@ class SVGPathSegCurvetoCubicSmoothRel final : public SVGPathSegCurvetoCubicSmoot
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoQuadraticSmoothAbs, CurveToQuadraticSmoothAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoQuadraticSmoothRel, CurveToQuadraticSmoothRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoCubicSmoothAbs, CurveToCubicSmoothAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoCubicSmoothRel, CurveToCubicSmoothRel);
```
Why do we need to add these traits? The removed code seems to work without it. I think createWrapper() uses `static_reference_cast<>`. Does the new binding generated code use `downcast<>`?

cdumez replied:
The generated bindings code uses `is<>()` / `downcast<>()`, which is why we need those traits. It needs it because it gets a base class pointer and it needs to figure out which subclass it actually is, in order to construct the right JS wrapper.

weinig reviewed (approved):
Sorry, missed this one.

cdumez committed:
Drop Custom toJS() / toJSNewlyCreated() functions for SVGPathSeg
https://bugs.webkit.org/show_bug.cgi?id=300288

Reviewed by Said Abou-Hallawa and Sam Weinig.

Drop Custom toJS() / toJSNewlyCreated() functions for SVGPathSeg now that
the bindings generator knows how to generate the right code.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/JSSVGPathSegCustom.cpp: Removed.
* Source/WebCore/svg/SVGPathSeg.h:
* Source/WebCore/svg/SVGPathSeg.idl:
* Source/WebCore/svg/SVGPathSegImpl.h:

Canonical link: https://commits.webkit.org/301171@main
